registry/registry.go

```diff
@@ -29,3 +29,14 @@ type Registry interface {
 	Records(zone string) ([]*endpoint.Endpoint, error)
 	ApplyChanges(zone string, changes *plan.Changes) error
 }
+
+//TODO(ideahitme): considering moving this to Plan
```

Comment:
Why don't you move it to endpoint?

You could have:

func (ep Endpoint) Owner() string {
    return ep.Labels[OwnerLabelKey]
}

type Endpoints []*Endpoint
func (eps Endpoints) FilterByOwner(ownerID string) Endpoints {
     filtered := Endpoints{}
     for _, ep := range eps {
             if ep.Owner() == ownerID {
                    filtered = append(filtered, ep)
             }
     }
    return filtered
}